File: app/src/main/java/com/example/autoqr/UserManagementActivity.java
package com.example.autoqr;

import android.app.AlertDialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.autoqr.adapters.UserAdapter;
import com.example.autoqr.models.User;
import com.example.autoqr.utils.UserManager;

import java.util.List;

public class UserManagementActivity extends AppCompatActivity {

    private ListView listViewUsers;
    private Button btnAddUser, btnClearAll;
    private UserAdapter userAdapter;
    private UserManager userManager;
    private List<User> userList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_management);

        initializeViews();
        initializeUserManager();
        setupClickListeners();
        loadUsers();
    }

    private void initializeViews() {
        listViewUsers = findViewById(R.id.listViewUsers);
        btnAddUser = findViewById(R.id.btnAddUser);
        btnClearAll = findViewById(R.id.btnClearAll);
    }

    private void initializeUserManager() {
        userManager = new UserManager(this);
    }

    private void setupClickListeners() {
        btnAddUser.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showAddUserDialog();
            }
        });

        btnClearAll.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showClearAllDialog();
            }
        });
    }

    private void loadUsers() {
        userList = userManager.getUsers();
        userAdapter = new UserAdapter(this, userList, new UserAdapter.UserActionListener() {
            @Override
            public void onEditUser(User user) {
                showEditUserDialog(user);
            }

            @Override
            public void onDeleteUser(User user) {
                showDeleteUserDialog(user);
            }

            @Override
            public void onToggleUserStatus(User user) {
                userManager.toggleUserStatus(user.getUserId());
                loadUsers();
                Toast.makeText(UserManagementActivity.this,
                        user.getName() + " status updated",
                        Toast.LENGTH_SHORT).show();
            }
        });
        listViewUsers.setAdapter(userAdapter);
    }

    private void showAddUserDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        LayoutInflater inflater = getLayoutInflater();
        View dialogView = inflater.inflate(R.layout.dialog_add_user, null);

        EditText etName = dialogView.findViewById(R.id.etName);
        EditText etUserId = dialogView.findViewById(R.id.etUserId);
        EditText etPassword = dialogView.findViewById(R.id.etPassword);

        builder.setView(dialogView)
                .setTitle("Add New User")
                .setPositiveButton("Add", null)
                .setNegativeButton("Cancel", (dialog, which) -> dialog.dismiss());

        AlertDialog dialog = builder.create();
        dialog.show();

        dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(v -> {
            String name = etName.getText().toString().trim();
            String userId = etUserId.getText().toString().trim();
            String password = etPassword.getText().toString().trim();

            if (validateInput(name, userId, password)) {
                if (userManager.addUser(name, userId, password)) {
                    Toast.makeText(this, "User added successfully", Toast.LENGTH_SHORT).show();
                    loadUsers();
                    dialog.dismiss();
                } else {
                    Toast.makeText(this, "User ID already exists", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    private void showEditUserDialog(User user) {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        LayoutInflater inflater = getLayoutInflater();
        View dialogView = inflater.inflate(R.layout.dialog_add_user, null);

        EditText etName = dialogView.findViewById(R.id.etName);
        EditText etUserId = dialogView.findViewById(R.id.etUserId);
        EditText etPassword = dialogView.findViewById(R.id.etPassword);

        etName.setText(user.getName());
        etUserId.setText(user.getUserId());
        etUserId.setEnabled(true);
        etPassword.setText(user.getPassword());

        builder.setView(dialogView)
                .setTitle("Edit User")
                .setPositiveButton("Update", null)
                .setNegativeButton("Cancel", (dialog, which) -> dialog.dismiss());

        AlertDialog dialog = builder.create();
        dialog.show();

        dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(v -> {
            String name = etName.getText().toString().trim();
            String password = etPassword.getText().toString().trim();

            if (validateInput(name, user.getUserId(), password)) {
                if (userManager.updateUser(user.getUserId(), name, password)) {
                    Toast.makeText(this, "User updated successfully", Toast.LENGTH_SHORT).show();
                    loadUsers();
                    dialog.dismiss();
                } else {
                    Toast.makeText(this, "Failed to update user", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    private void showDeleteUserDialog(User user) {
        new AlertDialog.Builder(this)
                .setTitle("Delete User")
                .setMessage("Are you sure you want to delete " + user.getName() + "?")
                .setPositiveButton("Delete", (dialog, which) -> {
                    if (userManager.deleteUser(user.getUserId())) {
                        Toast.makeText(this, "User deleted successfully", Toast.LENGTH_SHORT).show();
                        loadUsers();
                    } else {
                        Toast.makeText(this, "Failed to delete user", Toast.LENGTH_SHORT).show();
                    }
                })
                .setNegativeButton("Cancel", null)
                .show();
    }

    private void showClearAllDialog() {
        if (userList.isEmpty()) {
            Toast.makeText(this, "No users to clear", Toast.LENGTH_SHORT).show();
            return;
        }

        new AlertDialog.Builder(this)
                .setTitle("Clear All Users")
                .setMessage("Are you sure you want to delete all users? This action cannot be undone.")
                .setPositiveButton("Clear All", (dialog, which) -> {
                    userManager.clearAllUsers();
                    Toast.makeText(this, "All users cleared", Toast.LENGTH_SHORT).show();
                    loadUsers();
                })
                .setNegativeButton("Cancel", null)
                .show();
    }

    private boolean validateInput(String name, String userId, String password) {
        if (name.isEmpty()) {
            Toast.makeText(this, "Please enter name", Toast.LENGTH_SHORT).show();
            return false;
        }

        if (userId.isEmpty()) {
            Toast.makeText(this, "Please enter user ID", Toast.LENGTH_SHORT).show();
            return false;
        }

        if (password.isEmpty()) {
            Toast.makeText(this, "Please enter password", Toast.LENGTH_SHORT).show();
            return false;
        }

        if (userId.length() < 3) {
            Toast.makeText(this, "User ID must be at least 3 characters", Toast.LENGTH_SHORT).show();
            return false;
        }

        if (password.length() < 4) {
            Toast.makeText(this, "Password must be at least 4 characters", Toast.LENGTH_SHORT).show();
            return false;
        }

        return true;
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        finish();
    }
}

File: app/src/main/java/com/example/autoqr/UserManagementActivity.kt
package com.example.autoqr

import android.annotation.SuppressLint
import android.os.Bundle
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.combinedClickable
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.FilledTonalButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Surface
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.toArgb
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import androidx.compose.ui.res.stringResource
import androidx.core.view.WindowCompat
import com.example.autoqr.models.User
import com.example.autoqr.ui.theme.AutoqrTheme
import com.example.autoqr.utils.UserManager

class UserManagementActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            AutoqrTheme {
                Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colorScheme.background) {
                    val barColor = MaterialTheme.colorScheme.background.toArgb()
                    val dark = isSystemInDarkTheme()
                    SideEffect {
                        window.statusBarColor = barColor
                        window.navigationBarColor = barColor
                        val controller = WindowCompat.getInsetsController(window, window.decorView)
                        controller.isAppearanceLightStatusBars = !dark
                        controller.isAppearanceLightNavigationBars = !dark
                    }
                    UserManagementScreen(UserManager(this))
                }
            }
        }
    }
}

@SuppressLint("MutableCollectionMutableState")
@Composable
private fun UserManagementScreen(userManager: UserManager) {
    val context = LocalContext.current
    var users by remember { mutableStateOf(userManager.users) }
    var showAddDialog by remember { mutableStateOf(false) }
    var showClearAllConfirm by remember { mutableStateOf(false) }
    var editUser by remember { mutableStateOf<User?>(null) }
    var userToDelete by remember { mutableStateOf<User?>(null) }

    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        Text(
            text = stringResource(R.string.user_management),
            style = MaterialTheme.typography.headlineMedium,
            modifier = Modifier.padding(bottom = 8.dp)
        )
        LazyColumn(modifier = Modifier.weight(1f).padding(bottom = 16.dp)) {
            items(users) { user ->
                UserRow(
                    user = user,
                    onEdit = { toEdit -> editUser = toEdit },
                    onRequestDelete = { toDelete -> userToDelete = toDelete },
                    onToggle = { toToggle ->
                        userManager.toggleUserStatus(toToggle.userId)
                        users = userManager.users
                    }
                )
            }
        }

        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(12.dp)) {
            Button(onClick = { showAddDialog = true }, modifier = Modifier.weight(1f)) {
                Text(stringResource(R.string.add_user))
            }
            OutlinedButton(onClick = {
                if (users.isEmpty()) {
                    Toast.makeText(context, R.string.no_users_to_clear, Toast.LENGTH_SHORT).show()
                } else {
                    showClearAllConfirm = true
                }
            }, modifier = Modifier.weight(1f)) {
                Text(stringResource(R.string.clear_all))
            }
        }
    }

    if (showAddDialog) {
        showUserDialog(onDismiss = { showAddDialog = false }, onConfirmAdd = { name, userId, password ->
            val ok = userManager.addUser(name, userId, password)
            if (ok) {
                users = userManager.users
                showAddDialog = false
            } else {
                Toast.makeText(context, R.string.user_exists, Toast.LENGTH_SHORT).show()
            }
        })
    }

    editUser?.let { toEdit ->
        showUserDialog(
            initial = toEdit,
            onDismiss = { editUser = null },
            onConfirmEdit = { name, password ->
                if (userManager.updateUser(toEdit.userId, name, password)) {
                    users = userManager.users
                } else {
                    Toast.makeText(context, "Failed to update user", Toast.LENGTH_SHORT).show()
                }
                editUser = null
            }
        )
    }

    if (showClearAllConfirm) {
        AlertDialog(
            onDismissRequest = { showClearAllConfirm = false },
            title = { Text(stringResource(R.string.clear_all_users_title)) },
            text = { Text(stringResource(R.string.clear_all_users_message)) },
            confirmButton = {
                FilledTonalButton(onClick = {
                    userManager.clearAllUsers()
                    users = userManager.users
                    showClearAllConfirm = false
                }) { Text(stringResource(R.string.clear_all_users_button)) }
            },
            dismissButton = { OutlinedButton(onClick = { showClearAllConfirm = false }) { Text(stringResource(R.string.cancel)) } }
        )
    }

    userToDelete?.let { user ->
        AlertDialog(
            onDismissRequest = { userToDelete = null },
            title = { Text(stringResource(R.string.delete_user_title))},
            text = { Text(stringResource(R.string.delete_user_message, user.name)) },
            confirmButton = {
                FilledTonalButton(onClick = {
                    if (userManager.deleteUser(user.userId)) {
                        users = userManager.users
                    } else {
                        Toast.makeText(context, "Failed to delete user", Toast.LENGTH_SHORT).show()
                    }
                    userToDelete = null
                }) { Text(stringResource(R.string.delete)) }
            },
            dismissButton = { OutlinedButton(onClick = { userToDelete = null }) { Text(stringResource(R.string.cancel)) } }
        )
    }
}

@Composable
private fun UserRow(
    user: User,
    onEdit: (User) -> Unit,
    onRequestDelete: (User) -> Unit,
    onToggle: (User) -> Unit,
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .combinedClickable(
                onClick = { onEdit(user) },
                onLongClick = { onRequestDelete(user) }
            )
            .padding(vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Column(modifier = Modifier.weight(1f, fill = false).padding(end = 8.dp)) {
            Text(text = "${user.name} (${user.userId})", style = MaterialTheme.typography.titleMedium)
        }
        Switch(
            checked = user.isActive,
            onCheckedChange = { onToggle(user) }
        )
    }
}

@Composable
private fun showUserDialog(
    initial: User? = null,
    onDismiss: (() -> Unit)? = null,
    onConfirmAdd: ((String, String, String) -> Unit)? = null,
    onConfirmEdit: ((String, String) -> Unit)? = null,
) {
    var name by remember { mutableStateOf(initial?.name ?: "") }
    var userId by remember { mutableStateOf(initial?.userId ?: "") }
    var password by remember { mutableStateOf(initial?.password ?: "") }

    AlertDialog(
        onDismissRequest = { onDismiss?.invoke() },
        title = { Text(if (initial == null) stringResource(R.string.add_new_user) else stringResource(R.string.edit_user)) },
        text = {
            Column {
                OutlinedTextField(value = name, onValueChange = { name = it }, label = { Text(stringResource(R.string.full_name)) }, modifier = Modifier.fillMaxWidth())
                OutlinedTextField(value = userId, onValueChange = { userId = it }, label = { Text(stringResource(R.string.user_id)) }, modifier = Modifier.fillMaxWidth(), enabled = initial == null)
                OutlinedTextField(value = password, onValueChange = { password = it }, label = { Text(stringResource(R.string.password)) }, modifier = Modifier.fillMaxWidth())
            }
        },
        confirmButton = {
            Button(onClick = {
                if (name.isBlank()) { return@Button }
                if (initial == null) {
                    if (userId.isBlank() || userId.length < 3) { return@Button }
                    if (password.isBlank() || password.length < 4) { return@Button }
                    onConfirmAdd?.invoke(name.trim(), userId.trim(), password.trim())
                } else {
                    if (password.isBlank() || password.length < 4) { return@Button }
                    onConfirmEdit?.invoke(name.trim(), password.trim())
                    onDismiss?.invoke()
                }
            }) { Text(if (initial == null) stringResource(R.string.add) else stringResource(R.string.update)) }
        },
        dismissButton = { OutlinedButton(onClick = { onDismiss?.invoke() }) { Text(stringResource(R.string.cancel)) } }
    )
}
